pkg/jenkinsfile/pipeline.go

```diff
@@ -762,3 +791,206 @@ func (a *CreateJenkinsfileArguments) GenerateJenkinsfile(resolver ImportFileReso
 	}
 	return nil
 }
+
+// CreatePipelineSteps translates a step into one or more steps that can be used in jenkins-x.yml pipeline syntax.
+func (c *PipelineConfig) CreatePipelineSteps(step *syntax.Step, prefixPath string, args CreatePipelineArguments) ([]syntax.Step, int) {
+	steps := []syntax.Step{}
+
+	containerName := c.Agent.GetImage()
+
+	if step.GetImage() != "" {
+		containerName = step.GetImage()
+	}
+
+	dir := args.WorkspaceDir
+
+	if step.Dir != "" {
+		dir = step.Dir
+	}
+	dir = strings.Replace(dir, "/home/jenkins/go/src/REPLACE_ME_GIT_PROVIDER/REPLACE_ME_ORG/REPLACE_ME_APP_NAME", args.WorkspaceDir, -1)
```

Comment:
Yeah, I didn't put this in - but yes, that's what it's trying to do. I'll add a comment.

Reply:
...and the reason not to change the build packs at this point is that it'd break them for any existing Jenkinsfile Runner users.

Reply:
And Go is special because it uses a custom path in order to get have the source under a GOPATH structure? 

I guess if this is the only such "customization" we are fine for now, but if we would have several similar thing for other build packs as well, I'd try to generify this a bit.

Reply:
I guess I am most concerned about this invisible coupling between the Go buildpack and the code here.

Reply:
Yeah, I agree.